<a href="https://colab.research.google.com/github/SarahLares/introduction-to-deep-learning-with-keras-Coursera/blob/master/build_a_regression_model_in_keras_Part_A.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Regresioon Model in Keras

<div class="alert alert-block alert-info" style="margin-top: 20px">

<font size = 3>

In this course project, a regression model was constructed using the Keras deep learning library, and then experimented with increasing the number of training epochs and changing the number of hidden layers, and discussing how changing these parameters affects model performance. 

</font>
</div>

In [0]:
import pandas as pd
import numpy as np

Importing the data as a dataframe

In [2]:
concrete_data = pd.read_csv('https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DL0101EN/labs/data/concrete_data.csv')
concrete_data.head()

Cement  Blast Furnace Slag  Fly Ash  ...  Fine Aggregate  Age  Strength
0   540.0                 0.0      0.0  ...           676.0   28     79.99
1   540.0                 0.0      0.0  ...           676.0   28     61.89
2   332.5               142.5      0.0  ...           594.0  270     40.27
3   332.5               142.5      0.0  ...           594.0  365     41.05
4   198.6               132.4      0.0  ...           825.5  360     44.30

[5 rows x 9 columns]

So the first concrete sample has 540 cubic meter of cement, 0 cubic meter of blast furnace slag, 0 cubic meter of fly ash, 162 cubic meter of water, 2.5 cubic meter of superplaticizer, 1040 cubic meter of coarse aggregate, 676 cubic meter of fine aggregate. Such a concrete mix which is 28 days old, has a compressive strength of 79.99 MPa. 

Split data into predictors and target. The target variable in this problem is the concrete sample strength. Therefore, our predictors will be all the other columns.

In [0]:
concrete_data_columns = concrete_data.columns

predictors = concrete_data[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
target = concrete_data['Strength'] # Strength column

## A. Build a baseline model (5 marks)

<div class="alert alert-block alert-info" style="margin-top: 20px">

<font size = 3>

Use the Keras library to build a neural network with the following:

- One hidden layer of 10 nodes, and a ReLU activation function

- Use the adam optimizer and the mean squared error as the loss function.

</font>
</div>

In [3]:
import keras

Using TensorFlow backend.


### 1. Randomly split the data into a training and test sets by holding 30% of the data for testing. 

You can use the train_test_splithelper function from Scikit-learn.
